cfme/tests/configure/test_analysis_profiles.py

```diff
@@ -333,3 +345,72 @@ def test_analysis_profile_description_validation(analysis_profile_collection):
     )
     view.flash.assert_message("Description can't be blank")
     view.cancel.click()
+
+
+# Arguments of parametrize are product_features trees for Managing Quotas and Dialogs with RBAC
+@test_requirements.rbac
+@pytest.mark.tier(1)
+@pytest.mark.ignore_stream('5.9')
+@pytest.mark.parametrize(
+    'product_features',
+    PRODUCT_FEATURES_QUOTA + PRODUCT_FEATURES_DIALOG,
+    ids=["default_tenant", "custom_tenant", "add", "edit", "delete", "copy"]
+)
+def test_custom_role_modify_for_dynamic_product_feature(request, appliance, product_features):
+    """
+    Polarion:
+        assignee: ghubale
+        initialEstimate: 1/12h
+        caseimportance: high
+        caseposneg: positive
+        testtype: functional
+        startsin: 5.10
+        casecomponent: Configuration
+        tags: quota
+        testSteps:
+            1. create two tenants
+            2. create new custom role using existing role
+            3. Update newly created custom role by doing uncheck in to options provided under
+               automation > automate > customization > Dialogs > modify > edit/add/copy/delete
+               > uncheck for any tenant
+            4. Or Update newly created custom role by doing uncheck in to options provided under
+               Settings > Configuration > Access Control > Tenants > Modify > Manage Quotas
+               > uncheck for any tenant
+            5. You will see save button is not enabled but if you changed 'Name' or
+               'Access Restriction for Services, VMs, and Templates' then save button is getting
+               enabled.
+            6. It updates changes only when we checked or unchecked for all of the tenants under
+               edit/add/copy/delete options.
+        expectedResults:
+            1.
+            2.
+            3.
+            4.
+            5. 'save' button should be enabled after changing product feature tree.
+            6. It should work for individual tenant.
+
+    Bugzilla:
+        1655012
+    """
+    tenant = appliance.collections.tenants.create(
+        name=TENANT_NAME,
+        description="tenant_des{}".format(fauxfactory.gen_alphanumeric()),
+        parent=appliance.collections.tenants.get_root_tenant(),
+    )
+    request.addfinalizer(tenant.delete)
+    role = appliance.collections.roles.instantiate(name='EvmRole-tenant_quota_administrator')
+    copied_role = role.copy(name="{role}_{name}".format(
+        role=role.name, name=fauxfactory.gen_alpha()))
+    request.addfinalizer(copied_role.delete)
+    view = navigate_to(copied_role, 'Details')
+    # node_checked: Checks whether feature tree path is checked for given node
+    # Example:- Tuple: product_feature = (['Everything', 'Automation', 'Automate', 'Customization',
+    # 'Dialogs', 'Modify', 'Add', 'Add (tenant_27WyY6qCHi)'], False)
+    # 'False' argument with product_feature is required while updating copied_roles. It unchecks the
```

Comment:
Great explanation, but I don't think that should be in the tuple from the parametrization. It should be trivial to  inject it when you're actually trying to make the update. 

```
coped_role.update(
    {'product_features': [(f, False) for f in product_features]
)
```

Reply:
- Agree! I have taken product features as a list and then passed tuple to update the copied role: ```copied_role.update({'product_features': [(product_features, False)]})```
- Now I can able to remove indexing and passed ```product_features``` to ```node_checked```.